drivers/triton/triton.go

```diff
@@ -0,0 +1,573 @@
+package triton
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/log"
+	"github.com/docker/machine/state"
+	"github.com/docker/machine/utils"
+)
+
+const (
+	driverName                  = "triton"
+	TritonDefaultDockerPort     = 2376
+	TritonDefaultCloudapiDomain = "api.joyent.com"
+)
+
+type Driver struct {
+	CloudApiURL  string
+	DockerApiURL string
+	Account      string
+	PrivateKey   string
+	MachineName  string
+	storePath    string
+}
+
+func init() {
+	drivers.Register(driverName, &drivers.RegisteredDriver{
+		New:            NewDriver,
+		GetCreateFlags: GetCreateFlags,
+	})
+}
+
+func GetCreateFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:   "triton-url",
+			Usage:  "Triton Cloudapi URL",
+			Value:  "",
```

Comment:
API URLs generally have defaults in other drivers - I wonder if there's a sane default you could set here?